src/containers/Petitions.js

```diff
@@ -1,26 +1,76 @@
 import React from 'react';
 import Helmet from 'react-helmet';
 import { connect } from 'react-redux';
-import { fetchPetitions } from 'actions/PetitionsActions';
+import { withRouter } from 'react-router';
+import { petitionsPath } from 'helpers/petitionUrls';
+import { fetchPetitions, updateCityFilterValue } from 'actions/PetitionsActions';
 import settings from 'settings';
 import Petitions from 'components/Petitions';
 import getPetitions from 'selectors/petitions';
 
-const PetitionsContainer = React.createClass({
+const PetitionsContainer = withRouter(React.createClass({
   componentWillMount () {
     this.props.fetchPetitions(this.props);
   },
 
+  componentWillReceiveProps (nextProps) {
+    if (this.props.location.pathname !== nextProps.location.pathname) {
+      this.props.fetchPetitions(nextProps);
+    }
+  },
+
+  getAutocompleteProps () {
+    const router = this.props.router;
+    const updateCityFilterValue = this.props.updateCityFilterValue;
+    const cityFilterValue = this.props.cityFilterValue;
+
+    return {
+      name: 'city-filter',
+      endpoint: 'cities',
+      suggestionFormatter: (suggestion) => {
+        if (suggestion.name && suggestion.zips) {
+          return suggestion.name + ' - ' + suggestion.zips[0];
+        }
+
+        return '';
+      },
+      getFormValue: (suggestion) => {
+        return suggestion;
+      },
+      suggestionsLimit: 4,
+      helper: {
+        value: {
+          data: cityFilterValue
+        },
+        onChange (newValue) {
+          updateCityFilterValue(newValue);
+
+          if (newValue.id) {
+            router.push(petitionsPath({city: newValue}));
+          } else {
+            router.push(petitionsPath());
+          }
+        },
+        onBlur () {}
+      },
+      html: {
+        placeholder: 'Filter by city'
+      }
+    };
```

Comment:
This looks like there’s a lot of stuff going on in the return. Any chance to move a bit of the logic into helpers (e.g. the `suggestionFormatter` and `onChange`)?